backend/src/test/java/eu/bbmri_eric/negotiator/integration/api/v3/NegotiationPdfGenerationTest.java

```diff
@@ -0,0 +1,333 @@
+            .andExpect(
+                header()
+                    .string(
+                        "Content-Disposition",
+                        "attachment; filename=\"negotiation-" + NEGOTIATION_1_ID + ".pdf\""))
+            .andReturn();
+
+    byte[] pdfContent = result.getResponse().getContentAsByteArray();
+    assertNotNull(pdfContent);
+    assertTrue(pdfContent.length > 0);
+    assertTrue(new String(pdfContent).startsWith("%PDF"));
+  }
+
+  @Test
+  @WithMockNegotiatorUser(
+      id = 109L,
+      authorities = {"ROLE_ADMIN"})
+  void testGenerateNegotiationPdfWithAttachments_WithPdfAttachment_MergesPdfs() throws Exception {
+    MockMultipartFile pdfFile =
+        new MockMultipartFile("file", "test.pdf", "application/pdf", createSimplePdfBytes());
+
+    mockMvc
+        .perform(
+            multipart(ATTACHMENT_ENDPOINT, NEGOTIATION_1_ID)
+                .file(pdfFile)
+                .with(jwt().jwt(builder -> builder.subject("109"))))
+        .andExpect(status().isCreated());
+
+    MvcResult result =
+        mockMvc
+            .perform(
+                get(FULL_PDF_ENDPOINT, NEGOTIATION_1_ID)
+                    .with(jwt().jwt(builder -> builder.subject("109"))))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaType.APPLICATION_PDF))
+            .andExpect(
+                header()
+                    .string(
+                        "Content-Disposition",
+                        "attachment; filename=\"" + NEGOTIATION_1_ID + "_merged.pdf\""))
+            .andReturn();
+
+    byte[] mergedPdfContent = result.getResponse().getContentAsByteArray();
+    assertNotNull(mergedPdfContent);
+    assertTrue(mergedPdfContent.length > 0);
+    assertTrue(new String(mergedPdfContent).startsWith("%PDF"));
+  }
+
+  @Test
+  @WithMockNegotiatorUser(
+      id = 109L,
+      authorities = {"ROLE_ADMIN"})
+  void testGenerateNegotiationPdfWithAttachments_WithDocxAttachment_ConvertsAndMerges()
+      throws Exception {
+    MockMultipartFile docxFile =
+        new MockMultipartFile(
+            "file",
+            "test.docx",
+            "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+            createMinimalDocxBytes());
+
+    mockMvc
+        .perform(
+            multipart(ATTACHMENT_ENDPOINT, NEGOTIATION_1_ID)
+                .file(docxFile)
+                .with(jwt().jwt(builder -> builder.subject("109"))))
+        .andExpect(status().isCreated());
+
+    MvcResult result =
+        mockMvc
+            .perform(
+                get(FULL_PDF_ENDPOINT, NEGOTIATION_1_ID)
+                    .with(jwt().jwt(builder -> builder.subject("109"))))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaType.APPLICATION_PDF))
+            .andReturn();
+
+    byte[] mergedPdfContent = result.getResponse().getContentAsByteArray();
+    assertNotNull(mergedPdfContent);
```

Comment:
do these tests actually verify the merged pdf content? or just that a pdf was generated?